test/kb_fasttree_server_test.py

```diff
@@ -139,15 +182,49 @@ def test_kb_fasttree_run_FastTree_01(self):
             "gamma": "0",
         }
 
-        ret = self.getImpl().run_FastTree(self.getContext(), parameters)[0]
-        self.assertIsNotNone(ret["report_ref"])
-
-        # check created obj
-        report_obj = self.getWsClient().get_objects([{"ref": ret["report_ref"]}])[0]["data"]
-        self.assertIsNotNone(report_obj["objects_created"][0]["ref"])
-
-        created_obj_0_info = self.getWsClient().get_object_info_new(
-            {"objects": [{"ref": report_obj["objects_created"][0]["ref"]}]}
-        )[0]
-        self.assertEqual(created_obj_0_info[NAME_I], obj_out_name)
-        self.assertEqual(created_obj_0_info[TYPE_I].split("-")[0], obj_out_type)
+        # patch generate_run_id to return self.run_id
+        with patch("kb_fasttree.fasttree.generate_run_id", return_value=self.run_id):
+            ret = self.get_impl().run_FastTree(self.get_context(), parameters)[0]
+            assert ret["report_ref"] is not None
+
+            # check created obj
+            report_obj = self.get_ws_client().get_objects([{"ref": ret["report_ref"]}])[0]["data"]
+            assert report_obj["objects_created"][0]["ref"] is not None
+
+            created_obj_0_info = self.get_ws_client().get_object_info_new(
+                {"objects": [{"ref": report_obj["objects_created"][0]["ref"]}]}
+            )[0]
+            assert created_obj_0_info[1] == obj_out_name
+            assert created_obj_0_info[2].split("-")[0] == obj_out_type
+
+            # check that the expected files are in the expected places
```

Comment:
New test assertions below -- these check that the expected files are produced with the expected content. This allows the fasttree app code to be edited with at least a little confidence that any changes in output are deliberate.